src/object.jl

```diff
@@ -136,7 +136,7 @@ function _download_task(io)
     return nothing
 end
 
-mutable struct PrefechedDownloadStream{T <: Object} <: IO
+mutable struct PrefetchedDownloadStream{T <: Object} <: IO
```

Comment:
It would be great if you could add a docstring explaining:
- how this struct is meant to be used
- which API functions are thread safe (i think none of the public APIs are thread safe, and this is meant to be used from only one task, yeah? it would be great to make that explicitly clear.)
- etc